apps/api/src/domain/models/photograph/commands/create-photograph/photograph-created.event-handler.integration.spec.ts

```diff
@@ -0,0 +1,148 @@
+        },
+    })
+    .as({
+        id: photographId,
+        type: AggregateType.photograph,
+    })[0]; // There is only one event in this stream, which is the target event
+
+describe(`PhotographCreatedEventHandler`, () => {
+    let testQueryRepository: IPhotographQueryRepository;
+
+    let databaseProvider: ArangoDatabaseProvider;
+
+    // let arangoDatabaseForCollection: ArangoDatabaseForCollection<
+    //     IDetailQueryResult<IPhotographViewModel>
+    // >;
+
+    let app: INestApplication;
+
+    beforeAll(async () => {
+        const moduleRef = await Test.createTestingModule({
+            // providers: [CommandInfoService, PhotographCreatedEventHandler],
+            imports: [PersistenceModule.forRootAsync(), CommandModule, PhotographCommandsModule],
+        })
+            .overrideProvider(ConfigService)
+            .useValue(
+                buildMockConfigService(
+                    {
+                        ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+                    },
+                    buildConfigFilePath(Environment.test)
+                )
+            )
+            .compile();
+
+        await moduleRef.init();
+
+        app = moduleRef.createNestApplication();
+
+        await app.init();
+
+        const connectionProvider = app.get(ArangoConnectionProvider);
+
+        databaseProvider = new ArangoDatabaseProvider(connectionProvider);
+
+        // arangoDatabaseForCollection =
+        //     databaseProvider.getDatabaseForCollection('photograph__VIEWS');
+
+        testQueryRepository = new ArangoPhotographQueryRepository(
+            connectionProvider,
+            new ConsoleCoscradCliLogger()
+        );
+    });
+
+    beforeEach(async () => {
+        await databaseProvider.getDatabaseForCollection('photograph__VIEWS').clear();
+    });
+
+    afterAll(async () => {
+        databaseProvider.close();
+    });
+
+    describe(`when handling a photograph created event`, () => {
+        it(`should create the expected photograph`, async () => {
+            // const handler = app.get(PhotographCreatedEventHandler);
+
+            // @ts-expect-error Fix this issue
+            // await handler.handle(photographCreated);
+
+            await new PhotographCreatedEventHandler(testQueryRepository).handle(photographCreated);
+
+            const searchResult = await testQueryRepository.fetchById(photographId);
+
+            expect(searchResult).not.toBe(NotFound);
+
+            const foundPhotograph = searchResult as PhotographViewModel;
+
+            const { name: nameDto, contributions, actions } = foundPhotograph;
+
+            const name = new MultilingualText(nameDto);
+
+            const originalPhotographTitleItem = name.getOriginalTextItem();
+
+            expect(originalPhotographTitleItem.text).toBe(photographTitle);
+
+            expect(originalPhotographTitleItem.languageCode).toBe(languageCode);
+
+            expect(actions).toContain('TAG_RESOURCE');
```

Comment:
I wonder if now is the time to move the logic for actions to the view model itself.